### PR TITLE
Fix BTC order status bug

### PR DESCRIPTION
**Pull request by loon3**
Currently, orders with BTC remain in open status until expire_index is reached regardless of whether they're matched.  

The change to order.py will set a status of 'expired' for any BTC sell order that is part of an order match if a valid btcpay tx is not received within 20 blocks of the match.

The change to btcpay.py will set a status of 'filled' for any order that is part of an order match with BTC if a valid btcpay tx is received within 20 blocks of the match and either give_remaining or get_remaining is equal to zero.

### CODE REVIEW

File: counterpartylib/lib/messages/order.py [hunk 1]
```diff
@@ -426,6 +426,14 @@ def parse (db, tx, message):
 
         problems = validate(db, tx['source'], give_asset, give_quantity, get_asset, get_quantity, expiration, fee_required, tx['block_index'])
         if problems: status = 'invalid: ' + '; '.join(problems)
+
```
Comment (robby-dermody) [inline counterpartylib/lib/messages/order.py]: this should be in a different PR...seems like a separate issue

**Comment by loon3** [comment]
cancelling this PR to open three separate PRs for clarity
